modular_skyrat/modules/cargo/code/strappedcrate.dm

```diff
@@ -0,0 +1,102 @@
+			return
+		if(!user.transferItemToLoc(used_item, src))
+			to_chat(user, "<span class='warning'>For some reason, you can't attach [used_item]!</span>")
+			return
+		user.visible_message("<span class='notice'>[user] attaches [used_item] to [src].</span>", "<span class='notice'>You attach [used_item] to [src].</span>")
+		note = used_item
+		overlays += "manifest"
+
+/obj/structure/big_delivery/strapped/relay_container_resist_act(mob/living/user, obj/strapped_object)
+	if(ismovable(loc))
+		var/atom/movable/AM = loc //can't unwrap the wrapped container if it's inside something.
+		AM.relay_container_resist_act(user, strapped_object)
+		return
+	to_chat(user, "<span class='notice'>You lean on the back of [strapped_object] and start pushing to snap the straps off.</span>")
+	if(do_after(user, 150, target = strapped_object))  //takes slightly longer to break plastic as opposed to paper
+		if(!user || user.stat != CONSCIOUS || user.loc != strapped_object || strapped_object.loc != src )
+			return
+		to_chat(user, "<span class='notice'>You successfully broke [strapped_object]'s straps!</span>")
+		strapped_object.forceMove(loc)
+		playsound(src.loc, 'sound/items/poster_ripped.ogg', 50, TRUE)
+		new /obj/item/stack/sheet/plastic (loc, 4)
+		unwrap_contents()
+		qdel(src)
+	else
+		if(user.loc == src) //so we don't get the message if we resisted multiple times and succeeded.
+			to_chat(user, "<span class='warning'>You fail to remove [strapped_object]'s straps!</span>")
+
+//Lets plastic wrap crates and only crates, then overlays the plastic straps/locks it shut
+//Basically applies package wrapping code but for plastic onto crates
+/obj/item/stack/sheet/plastic/afterattack(obj/target, mob/user, proximity)
+	. = ..()
+	if(!proximity)
+		return
+	if(!istype(target))
+		return
+	if(target.anchored)
+		return
+
+	if(istype (target, /obj/structure/closet))	//this extra IF check is basically so that hitting EVERY item doesnt say 'you cant wrap this', and only hitting invalid storage items does
+		if(istype (target, /obj/structure/closet/crate))
+			var/obj/structure/closet/crate/tcrate = target
+			if(tcrate.opened)
+				return
+			if(!tcrate.delivery_icon) //no delivery icon means unwrappable crate
+				to_chat(user, "<span class='warning'>You can't strap this shut!</span>")
+				return
+			if(istype(tcrate, /obj/structure/closet/crate/coffin) || istype(tcrate, /obj/structure/closet/crate/trashcart) || istype(tcrate, /obj/structure/closet/crate/bin) || istype(tcrate, /obj/structure/closet/crate/critter) || istype(tcrate, /obj/structure/closet/crate/necropolis) || istype(tcrate, /obj/structure/closet/crate/miningcar))	//for buggy-wuggy and ugly-wugly crates
+				to_chat(user, "<span class='warning'>You can't strap this shut!</span>")
+				return
+			if(use(4))
+				if(!do_after(user, 20, target = user))	//For some reason this still deletes stacks of exactly four, even if you fail?
```

Comment:
The reason it's deleting the stacks even though you fail is because use(4) is where it actually deletes them. Move this check up, and nest the if(use) statement into the do after.